### ISSUE TITLE
[Feature request] window.json variable

### ISSUE THREAD

**Issue by KaKi87** [pd4d10/json-viewer#11]
Hello,

Would it be possible to allow access to the inspected JSON data from devtools using a `window.json` variable, like nikrolls/json-formatter does ?

Thanks !

**Comment by KaKi87** [comment]
up, thanks

**Comment by st-jordy-de-koning** [comment]
Bit of a workaround, but you can currently extract the raw JSON data from the page's #rawdata-panel container. Should help when trying to test code from the console.

const json = JSON.parse(document.querySelector('#rawdata-panel pre.data').innerHTML)

**Comment by KaKi87** [comment]
@st-jordy-de-koning Interesting !

@pd4d10 Any news on this ?

Thanks

**Comment by pd4d10** [comment]
Thanks for the suggestion!

We are going to upgrade to [manifest v3](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-overview/). There seems to be a technical limitation to inject global variables after this upgrade, due to the CSP policy.

The extension you mentioned seems still stay at manifest v2, which has already been deprecated by Chrome officially.

**Comment by KaKi87** [comment]
I'll look into that.

If creating a `window.json` variable turns out to be truly impossible, I would like to suggest adding a button to the UI which, on click, would copy a `const json` code snippet to clipboard, that could be pasted into DevTools.

I'm also wondering if an alternative extensions store will some day exist, since there's no way Chromium-based browser users like us give up on adblockers anyway.

**Comment by pd4d10** [comment]
> a button to the UI which, on click, would copy a `const json` code snippet to clipboard, that could be pasted into DevTools.

<img width="123" alt="image" src="https://user-images.githubusercontent.com/9524411/171991420-fad76d7e-9da1-4b68-8beb-36d765998a95.png">

Perhaps the `Copy` button is the feature you want

**Comment by KaKi87** [comment]
No, only copies the JSON value, but it will still require to write `const json = ''` into DevTools.

I assure you, automating this is a huge gain of time.

**Comment by pd4d10** [comment]
OK get your point.

The original purpose of this extension is to port Firefox's corresponding JSON Viewer, so we are trying not to add too many extra features to keep it consistent with the original one.

But I also agree that it's useful if someone wants to do some debug in the console.

**Comment by KaKi87** [comment]
Hello,

Any news on this ?

I tried to add this feature as a userscript, but the `#rawdata-panel` element remains empty until the `Raw Data` button is clicked, meanwhile I won't do something as dirty as automatically clicking on it then clicking back on the `JSON` button.

Thanks

**Comment by KaKi87** [comment]
@pd4d10 Please ? Thanks

**Comment by PythonicNinja** [comment]
Any news?

**Comment by KaKi87** [comment]
Dead project
